File: 0290-word-pattern/0290-word-pattern.cpp
class Solution {
public:
    bool wordPattern(string pattern, string s) {
    vector<string>ss;
    string temp; s.push_back(' ');
    unordered_map<char,string>matchChar;
    set<string>matchWord;
    for (char i : s) {
        if (i != ' ')
            temp+=i;

        else{
            ss.push_back(temp);
            temp="";
        }
    }
    if (ss.size() != pattern.size())
        return false;
    for (int i = 0; i < pattern.size(); ++i) {
        if (matchChar.find(pattern[i]) != matchChar.end()){
            if(matchChar[pattern[i]] != ss[i] )
                return false;
        }
        else{
            if(matchWord.count(ss[i])==1)
                return false;
            matchChar[pattern[i]]=ss[i];
            matchWord.insert(ss[i]);
        }
    }
    return true;
}
};

File: 0290-word-pattern/0290-word-pattern.cs
public class Solution {
    public bool WordPattern(string pattern, string s) {
        var words = s.Split();
        HashSet<string> strings = new(words);
        HashSet<char> chars = new(pattern.ToCharArray());
        if (pattern.Length != words.Length || strings.Count != chars.Count) return false;
        Dictionary<char, string> map = new();
        for (int i = 0; i < pattern.Length; i++)
        {
            if (map.ContainsKey(pattern[i]))
            {
                if (map[pattern[i]] != words[i])
                    return false;
            }
            else
                map[pattern[i]] = words[i];
        }
        return true;
    }
}